5.6.35 maintenance modasdata

dnaldoog opened the pull request:
Add setModulatorValuesFromData and getModulatorValuesFromData functions to incomplete RK implementation of single byte message creation from ModulatorValues.

1. Supports 2 byte MSB/LSB or LSB/MSB 7 bit sysex message creation from Modulator (mapped) Values
2. Supports 2 byte msb/lsb or lsb/msb 4 bit sysex message creation from Modulator (mapped)  Values
3. Supports 2 byte msb/lsb or lsb/msb 4 bit signed 8bit sysex message creation from Modulator (mapped)  Values (-128 - +127)

----

1. Supports 1 byte  sysex message data assignment from incoming MIDI to Modulators
2. Supports 2 byte MSB/LSB or LSB/MSB 7 bit sysex message data assignment from incoming MIDI to Modulators
3. Supports 2 byte msb/lsb or lsb/msb 4 bit sysex message data assignment from incoming MIDI to Modulators
4. Supports 2 byte msb/lsb or lsb/msb 4 bit signed 8bit sysex message data assignment from incoming MIDI to Modulators

Example panel: https://github.com/dnaldoog/CtrlrDemoPanels/blob/main/Panels/DEMO/DEMO%20-%20Data%20dump%20utilities_3_0.panel

damiensellier committed:
5.6.35 UPDATED CtrlrAutoSave & CtrlrAutoSaveInterval only if isStandaloneApp

damiensellier committed:
5.6.35 UPDATED Version from 5.6.34 to 5.6.35 - CMakeLists IS_MIDI_EFFECT FALSE

* **UPDATED** Version from 5.6.34 to 5.6.35
* **UPDATED** CMakeLists IS_MIDI_EFFECT FALSE

dnaldoog committed:
Ctrlr Linux uncomment to compile

damiensellier commented:
Hi John, are you sure about the linux ifdef ? I never had any problem as it was?

https://github.com/damiensellier/CtrlrX/blob/a2f6c451557fe540444dd37fa0b7296c4bae717a/Source/Native/CtrlrLinux.cpp#L2

dnaldoog commented:
You can uncomment it - Before I had to comment it out in order to compile on VS as it was including CtrlrLinux.cpp and failing.

I get these compile errors 
<img width="277" height="340" alt="{61FED5D5-03A5-429F-9334-D64ADBA25800}" src="https://github.com/user-attachments/assets/5517fcc1-b58a-4848-811b-4e3f5e40fb30" />

I asked about another way to do that and this was suggested by Gemini. 
This doesn't seem to work though.

#ifdef __linux__
    #define LINUX            // Defines your custom flag (if you use it elsewhere)
    #define PACKAGE "Ctrlr"  // Defines the package name
#endif

dnaldoog commented:
Also I think libr-bfd.c and libr.c need to be removed from projucer files.

<img width="228" height="169" alt="{F23CA10D-DFCF-4079-B3A4-37B426C484A9}" src="https://github.com/user-attachments/assets/30a96749-4fb1-49ec-8562-7ed068d68e27" />

dnaldoog committed:
Added modasdata back in

dnaldoog committed:
Clear tabId add GROUP when moving component

dnaldoog committed:
update CtrlrLuaManager

dnaldoog commented:
This works on Windows in VisualStudio

````
#include "stdafx.h"
#ifdef __linux__
#define PACKAGE "Ctrlr"
#include "CtrlrMacros.h"
#include "CtrlrLinux.h"
````

not ..

````
#include "stdafx.h"
#define LINUX
#ifdef LINUX
#define PACKAGE "Ctrlr"

#include "CtrlrMacros.h"
#include "CtrlrLinux.h"
````

dnaldoog committed:
update CtrlrLuaManager lua_pushcfunction-luaStateAudio

dnaldoog committed:
Merge branch '5.6.35-Maintenance_modasdata' of https://github.com/dnaldoog/CtrlrX into 5.6.35-Maintenance_modasdata
(1) Removed libopen_usb from CtrlrLuaManager  (2) Copy-paste bug in createLuaStateAudio()

dnaldoog committed:
createLuaStateAudio fix and luaopen_usb in CtrlrLuaManager

dnaldoog commented:
After removing the constructor for initialising a MemoryBlock, removed due to panels and users using getSize() thus doubling File data sizes, I have implemented two MemoryBlock initialise methods
````
>>> m=MemoryBlock()

>>> m:init(33)

>>> console(m:toHexString(1))
00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00

>>> console(String(m:getSize()))
33

-- static version
>>> z=MemoryBlock.createWithSize(44)

>>> console(String(z:getSize()))
44
````

https://github.com/dnaldoog/CtrlrX/blob/9a17c15bcbb77b1dc322729ed3b610c1cdb4e2f1/Source/Lua/JuceClasses/LMemory.cpp#L483

Commit:
removed duplicate luabindings frim merge edit

damiensellier commented:
I have errors everywhere from the changes you submited for the modasdata . especially on the Lmemory file and memoryblock

BTW what is the right thing to add in the projucer linux library field if we want to remove the useless ones, right now it's : 

```
udev
:libbfd.a
iberty
z
X11
sframe
zstd
```

which ones should we removed ?

dnaldoog commented:
The only files I removed were libr_bfd.c and libr.c


Are you using 
https://github.com/dnaldoog/CtrlrX/tree/5.6.35-Maintenance_modasdata?tab=readme-ov-file

Can you post some errors?

dnaldoog commented:
Last night in a commit, there were many errors resulting from a botched merge, but I am pretty sure I fixed all that so the latest commit should be working. I will check soon.

dnaldoog commented:
The only files I made major changes to were LMemory and  CtrlrLuaManager and CtrlrMacros.h. Also the one liners to uiGroup/tabs fix and the ifdef for Linux.

damiensellier commented:
Thanks, I'm slowly incorporating the different commits we both did at the same time. I'll work on a branch and go slowly until it breaks. I wonder if all the error were coming from the fact projucer was still calling the Linux libs we removed with cmakelists. I did a commit to push all updated projucer files.

For the moment 5.6.35-Maintenance is working great. 

Next step is to implement the get/setmodasvalue and go from there. I'm not that far away.

dnaldoog commented:
I just did a complete new clone and build of that branch in Windows and it is compiling no problem. Also the projucer files in that branch have already been edited to remove libr_bfd.c and libr.c

The new getModulatorValuesAsData is working too.

<img width="703" height="496" alt="{24538736-6694-4638-AB8C-7E3F425C45CA}" src="https://github.com/user-attachments/assets/fbfc795f-935a-411d-b6d9-a6f452ceffea" />

dnaldoog commented:
There are a couple of issues with `getModulatorValuesAsData` and `getModulatorValuesAsData` I thought were resolved, but aren't so I am going to do another commit to that branch when I fix the problem(s)

dnaldoog committed:
modulatorCustomIndex 0 int/string confusion

dnaldoog committed:
setModulatorValuesAsData refactored

dnaldoog committed:
Added stricter 0 checking but  maybe cached issue on 0  modindex val?"

dnaldoog commented:
I'm pretty sure I've got it working now (See latest commits)

There was a problem parsing `modulatorCustomIndex `when it is 0 (i.e. the first modulator in a series) - it would be easier to perhaps make the index start at 1 - for now though if it is working, we can leave it as it is.

Here is a demo panel for you to test each mode:
https://github.com/dnaldoog/CtrlrDemoPanels/blob/main/Panels/DEMO/DEMO%20-%20Data%20dump%20utilities%20live%203.panel

It actually could become quite a **central feature of CtrlrX** as it would obviate all those lua tables and gathering of modulator data that is usually needed on a complex panel, especially when users need to create 7bit or 4 bit splits etc - all of that would be automatic.

Maybe Roman  originally intended this functionality to be central to panel development, but it was never finished?

<img width="571" height="429" alt="{76CD5408-C085-4E0E-AE8A-741FDCBA9087}" src="https://github.com/user-attachments/assets/268d92c7-22e0-4b4a-959c-7bd2aa6b4b07" />

damiensellier commented:
Just to recap, what are the files you modded to get this working? it's hard to track because again I have to go through all your commits and check them one by one :

Source/Core/CtrlrMacros.h
Source/Core/CtrlrPanel/CtrlrPanel.cpp
Source/Lua/CtrlrLuaManager.cpp

Please confirm 
I'm probably going to just copy paste yours from your most recent branch, so I want to be sure I don't miss anything.


For info, this is where I'm at :  https://github.com/damiensellier/CtrlrX/commits/5.6.35-Maintenance/

I miss the new linux macro ifdef __LINUX__  I also miss the update about the libsframe so 2 in the cmake file and the CI yml script (https://github.com/damiensellier/CtrlrX/tree/5.6.35-Maintenance-Linux-libsframe-so-2-FIX) but I have all the others I think.

dnaldoog commented:
This seems right for the set/getModulatorValuesAsData.

dnaldoog commented:
In CtrlLuaManager it’s all the code from line 745 down to the lua binding wrapForLua code plus some enum additions line 1380. On iPhone so difficult to give details.

damiensellier commented:
UPDATED on 2025.11.24 for CtrlrX 5.6.35 commit 42307ac3c5d20ab18484d263e35216ee43b92619
